setup.py

```diff
@@ -413,6 +413,12 @@ def run(self):
         else:
             report('-- Building without distributed package')
 
+        # Do not use clang to compile exensions if `-fstack-clash-protection` is defined
+        # in system CFLAGS
+        system_c_flags = distutils.sysconfig.get_config_var('CFLAGS')
+        if IS_LINUX and '-fstack-clash-protection' in system_c_flags and 'CC' in os.environ and 'clang' in os.environ['CC']:
```

Comment:
Minor simplification:

    'clang' in os.environ.get('CC', '')

instead of:

    'CC' in os.environ and 'clang' in os.environ['CC']